Doc/library/importlib.rst

```diff
@@ -1327,69 +1326,77 @@ find and load modules.
 .. class:: ModuleSpec(name, loader, *, origin=None, loader_state=None, is_package=None)
 
    A specification for a module's import-system-related state.  This is
-   typically exposed as the module's ``__spec__`` attribute.  In the
+   typically exposed as the module's :attr:`__spec__` attribute.  In the
    descriptions below, the names in parentheses give the corresponding
-   attribute available directly on the module object.
-   E.g. ``module.__spec__.origin == module.__file__``.  Note however that
-   while the *values* are usually equivalent, they can differ since there is
-   no synchronization between the two objects.  Thus it is possible to update
-   the module's ``__path__`` at runtime, and this will not be automatically
-   reflected in ``__spec__.submodule_search_locations``.
+   attribute available directly on the module object, e.g.
+   ``module.__spec__.origin == module.__file__``.  Note, however, that while
+   the *values* are usually equivalent, they can differ since there is no
+   synchronization between the two objects.  For example, it is possible to
+   update the module's :attr:`__file__` at runtime and this will not be
+   automatically reflected in the module's :attr:`__spec__.origin`, and vice
+   versa.
 
    .. versionadded:: 3.4
 
    .. attribute:: name
 
-   (``__name__``)
+   (:attr:`__name__`)
 
-   A string for the fully-qualified name of the module.
+   The module's fully-qualified name.
+   The :term:`finder` should always set this attribute to a non-empty string.
 
    .. attribute:: loader
 
-   (``__loader__``)
+   (:attr:`__loader__`)
 
-   The :term:`Loader <loader>` that should be used when loading
-   the module.  :term:`Finders <finder>` should always set this.
+   The :term:`loader` used to load the module.
+   The :term:`finder` should always set this attribute.
 
    .. attribute:: origin
 
-   (``__file__``)
+   (:attr:`__file__`)
 
-   Name of the place from which the module is loaded, e.g. "builtin" for
-   built-in modules and the filename for modules loaded from source.
-   Normally "origin" should be set, but it may be ``None`` (the default)
-   which indicates it is unspecified (e.g. for namespace packages).
+   The location from which the module was loaded (usually the path).
+   The :term:`finder` should always set this attribute but it may be ``None``
+   when unspecified (like for namespace packages).
 
    .. attribute:: submodule_search_locations
 
-   (``__path__``)
+   (:attr:`__path__`)
 
-   List of strings for where to find submodules, if a package (``None``
-   otherwise).
+   The list of paths where to find submodules within a package.
```

Comment:
As mentioned above, it doesn't have to be dirnames.  Also, `module.__path__` is what the import system uses so we only need to include details here relevant to Finder/Loader authors (instead of users).

```suggestion
   The list of places the import system should look for submodules.
   (See :attr:`__path__` for more info on how it is used.)
```

Reply:
Updated.